### PR TITLE
[docs] [Query Performance]

### PR DESCRIPTION
**Pull request by minhtuev**
## What changes are proposed in this pull request?

Added documentation for Query Performance panel

## How is this patch tested? If it is not, please explain why.

* Local testing

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [x] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
	- Introduced a "Query Performance" section in the documentation, detailing enhancements for sidebar and background queries.
	- Added a toast notification system to alert users about potential query performance improvements.
	- Users can create and manage indexes and summary fields through the new Query Performance panel.

- **Documentation**
	- Enhanced FiftyOne Teams documentation for better navigation with a new entry in the table of contents.
	- Updated FiftyOne App documentation to clarify index management for Teams customers, highlighting the use of the built-in Query Performance panel.

### CODE REVIEW

File: docs/source/teams/query_performance.rst [hunk 1]
```diff
@@ -0,0 +1,123 @@
+.. _query-performance:
+
+Query Performance (NEW)
+=======================
+
+Query Performance is a feature built into the :ref:`FiftyOne Teams App <teams-app>`
+which allows users to use FiftyOne to improve the performance of sidebar and background
+queries through the use of indexes and summary fields. Query Performance subsumed and
+expanded the capabilities of the previous Fiftyone's Lightning Mode (LM).
```
Comment (benjaminpkane) [inline docs/source/teams/query_performance.rst:9]: I think LM mentions can be omitted
```suggestion
queries through the use of indexes and summary fields.
```

**Review by mdecarlo — CHANGES_REQUESTED** [review]
A few grammatical changes, and suggestions to remove certain paragraphs.

File: docs/source/teams/query_performance.rst [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+Query Performance Toast
+-----------------------
+
+When you open the FiftyOne Teams App with Query Performance enabled, you will see a toast
+notification whenever a query is run that could benefit from Query Performance. The query
+can be part of the background sampling process or a query that you run manually from the
+side bar. Sidebar queries are queries that you run by clicking on the sidebar filters; for
+example you can click on a label filter on the sidebar, and if the filter takes too longer
+the toast will be opened.
```
[the remark this answers is not in the document]
Comment (minhtuev) [reply docs/source/teams/query_performance.rst:54]: Do we need to document the Dismiss behavior?

File: docs/source/user_guide/app.rst [hunk 3]
```diff
@@ -575,9 +575,8 @@ datasets:
 
 .. note::
 
-    Did you know? You can manage dataset indexes via the App by installing the
-    `@voxel51/indexes <https://github.com/voxel51/fiftyone-plugins/tree/main/plugins/indexes>`_
-    plugin!
+    Did you know? Teams customer can manage dataset indexes via the App by with the builtin Query Performance panel!
+    See :ref:`this page <query-performance>` for more information.
```
[the remark this answers is not in the document]
Comment (CamronStaley) [reply docs/source/user_guide/app.rst:579]: the bot is correct

**Review by CamronStaley — DISMISSED** [review]

**Review by CamronStaley — APPROVED** [review]

**Pull request merged**